#! /bin/sh /usr/share/dpatch/dpatch-run
## fix_tree_copies.dpatch by Sheldon Hearn <sheldonh@starjuice.net>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix tree copies (upstream r1375)

@DPATCH@

--- fsvs-1.1.12.orig/tests/044_copyfrom
+++ fsvs-1.1.12/tests/044_copyfrom
@@ -227,3 +227,9 @@
 
 $WC2_UP_ST_COMPARE
 
+sleep 1
+cp -r a with_subs
+$BINq cp a with_subs
+$BINq ci -m "subs"
+
+$WC2_UP_ST_COMPARE
--- fsvs-1.1.12.orig/src/waa.c
+++ fsvs-1.1.12/src/waa.c
@@ -2487,11 +2487,9 @@
 	to_append=NULL;
 	status=0;
 
+	ops__copy_single_entry(src, dest);
 	if (!S_ISDIR(src->st.mode))
-	{
-		ops__copy_single_entry(src, dest);
 		goto ex;
-	}
 
 
 	append_count=0;
--- fsvs-1.1.12.orig/CHANGES
+++ fsvs-1.1.12/CHANGES
@@ -1,3 +1,7 @@
+Changes since 1.1.12
+- Bugfix for directory structure copying committed .. should have been in 
+	1.1.12.
+
 Changes since 1.1.11
 - Licence change to GPLv3.
 - Some cleanups
